tests/base/src/test/java/org/keycloak/test/admin/SMTPConnectionTest.java

```diff
@@ -109,22 +149,26 @@ public void testWithAuthEnabledCredentialsEmpty() throws Exception {
 
     @Test
     public void testWithAuthEnabledValidCredentials() throws Exception {
-        greenMailRule.credentials("admin@localhost", "admin");
+        String password = "admin";
+
+        mailServer.credentials("admin@localhost", password);
         Response response = realm.testSMTPConnection(settings("127.0.0.1", "3025", "auto@keycloak.org", "true", null, null,
-                "admin@localhost", SMTP_PASSWORD));
+                "admin@localhost", password));
         assertStatus(response, 204);
     }
 
     @Test
     public void testAuthEnabledAndSavedCredentials() throws Exception {
```

Comment:
This can be cleaned up a bit by using `ManagedRealm#updateWithCleanup(r -> r.smtp(...))` - that will remove the new to keep the new/old rep, try/finally, etc.